### PR TITLE
Ease maintenance

### LINKED ISSUE #235: Maintenance-related issues
I'm testing this over OTP 19 to 23, so instead of creating several individual issues I'll try to concentrate what I find important in this one.

___1
Q: Is `rebar3_codecov` actively used?
I know the plugin is in `rebar.config` but the associated project seems unmaintained and there's already a compilation warning (not very important).

Targets:
* OTP 19
* OTP 20 (also shows a warning on `export_all`)
* OTP 21 (also shows warning on `erlang:get_stacktrace/0: deprecated`)
* OTP 22
* OTP 23 (also shows a warning on `===> Unable to run post hooks for 'ct', command 'analyze' in namespace 'codecov' not found.`)

**Proposal**: get rid of `rebar3_codecov`. (this also prevents running tests on OTP 19, since it's using yet-undefined `string:split`)

___2
Q: Is there a replacement for `aleppo` we want to adopt?
There's compilation warnings when doing `rebar3 ct`.

Targets:
* OTP 19
* OTP 20
* OTP 21
* OTP 22
* OTP 23

**Proposal**: none, waiting on your feedback.

___3
Q: Is there a replacement for `katana` (0.4.0) we want to adopt?
There's compilation warnings when doing `rebar3 ct`.

Targets:
* OTP 20
* OTP 21 (also shows warning on `erlang:get_stacktrace/0: deprecated`)
* OTP 22
* OTP 23

**Proposal**: none, waiting on your feedback.

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
Closes #235.

I'm pushing now to get some guidance on requirements moving forward.

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Travis CI does not seem activated for this project. Is this something you could look at? It would make sense, since `.travis.yml` is already there, and all.

File: .travis.yml
Comment (elbrujohalcon) [inline .travis.yml:8]: We can totally drop support for `19.3`. I would even argue that we can drop support for `20.3`, too.

**Review by elbrujohalcon — APPROVED** [review]
Let's wait for the release after inaka/erlang-katana#160 is merged, but this looks good.

File: .travis.yml
Comment (paulo-ferraz-oliveira) [reply .travis.yml:8]: 😄 I wouldn't, though, but if you prefer to remove it from `.travis.yml` alone (e.g. not mess with `minimum_otp_vsn` in `rebar.config`), sure, you're the bawss.

**Note**: I like to keep my "versions" with the same schedule as `rebar3` (so it's 19+, for now).

Comment (elbrujohalcon) [reply .travis.yml:8]: Our policy (when Inaka was a company) was to support only the last 2 versions of OTP. If you work with older versions of OTP, use older releases of our libs. And if you need new features in them, just ask us… and we backport them.

Comment (elbrujohalcon) [reply .travis.yml:8]: We backported stuff a total of… one single time through our whole 7 years of existence. 🤷

Comment (paulo-ferraz-oliveira) [reply .travis.yml:8]: I understand you don't want to support 19.3 or even more recent versions (not everybody has the luxury of constantly moving a few hundred libs **up** all the time - and when the deps go deep, it becomes a real hassle).

Let me know, so I have to **not** care. And tell me explicitly which ones you want to support, so I know.

Comment (elbrujohalcon) [reply .travis.yml:8]: I would be pragmatic here: leave all there until we make a change that breaks the old ones. If/when that's the case, instead of forcing ourselves to write backwards-compatible code, just increase minimum_otp_vsn in rebar.config.

The perfect example of this are stack traces. I would not support `erlang:get_stacktrace()` and `Kind:Reason:Stack` in the same codebase. Once we move forward... We entirely remove support for the previous stuff.

Comment (elbrujohalcon) [reply .travis.yml:8]: But this is @ferigis' repo. He gets to decide what to do here, not me.

Comment (paulo-ferraz-oliveira) [reply .travis.yml:8]: > leave all there

Exactly what I was suggesting. Don't bump for the sake of bumping (though I do understand there's an "implied" cost in maintenance), and keep testing until something needs to break in a given past version, out of necessity.

Comment (paulo-ferraz-oliveira) [reply .travis.yml:8]: > The perfect example of this are stack traces. I would not support erlang:get_stacktrace() and Kind:Reason:Stack in the same codebase. Once we move forward... We entirely remove support for the previous stuff.

Sure, but until you don't, you'll get warnings (which some might not find acceptable, in their context, so have to fork or stay in a previous version), and when you do it might be very abrupt, since you're relying on OTP to not be abrupt.

In any case, it's always a maitainer's choice and can be imposed via `rebar.config`'s `minimum_otp_vsn`, for example, so it's clear.

**Review by elbrujohalcon — APPROVED** [review]

**Comment by elbrujohalcon** [comment]
@ferigis should merge, but I already approved 😏

**Comment by paulo-ferraz-oliveira** [comment]
@ferigis: ping

**Review by ferigis — APPROVED** [review]

**Comment by ferigis** [comment]
hi, I can't merge it if travis is complaining

**Comment by paulo-ferraz-oliveira** [comment]
Let me see what's stopping it.

**Comment by paulo-ferraz-oliveira** [comment]
That's odd. It seems to not find katana 1.0.0

**Comment by paulo-ferraz-oliveira** [comment]
I'm bumping `rebar3` (in script `ci`) to "latest available", since this seems to be the origin of the failing CI checks.

**Comment by paulo-ferraz-oliveira** [comment]
I see this failing, now, only for OTP 20.3. I don't know if that's good or bad. :(
@ferigis, do you want me to look into this?

**Comment by elbrujohalcon** [comment]
…aaaand merged!

**Pull request merged**
